lib/packagekit-glib2/pk-client.c

```diff
@@ -1965,15 +1964,20 @@ pk_client_create_helper_socket (PkClientState *state)
 
 	/* either the self test failed, or we're not in self test */
 	if (!ret) {
-		ret = pk_client_create_helper_argv_envp (state,
-							 &argv,
+		ret = pk_client_create_helper_argv_envp (&argv,
 							 &envp);
 	}
 
 	/* no supported frontends available */
 	if (!ret)
 		return NULL;
 
+	socket_filename = g_build_filename(g_get_user_runtime_dir(), "debconf-socket", NULL);
+	if (g_file_test(socket_filename, G_FILE_TEST_EXISTS))
+		goto out;
+
+	g_free(socket_filename);
```

Comment:
This is a bit dangerous - socket_filename is now pointing to invalid memory. You should make use a new variable name for the debconf-socket string to avoid a bug being introduced in the future. The other option would be to use `g_clear_pointer (&socket_filename, g_free)` to set it to NULL (I still think multiple variables is clearer and safer).